domain/src/main/java/camp/nextstep/edu/calculator/domain/Operator.kt

```diff
@@ -0,0 +1,64 @@
+package camp.nextstep.edu.calculator.domain
+
+enum class Operator(val operator: String) {
+    PLUS("+") {
+        override fun calculate(preNumber: String, postNumber: String): String {
+            if (!preNumber.isNum() || !postNumber.isNum()) {
+                throw IllegalArgumentException("Parameter must be Number")
+            }
+            return (preNumber.toInt() + postNumber.toInt()).toString()
+        }
+    },
+    MINUS("-") {
+        override fun calculate(preNumber: String, postNumber: String): String {
+            if (!preNumber.isNum() || !postNumber.isNum()) {
+                throw IllegalArgumentException("Parameter must be Number")
+            }
+            return (preNumber.toInt() - postNumber.toInt()).toString()
+        }
+    },
+    TIMES("*") {
+        override fun calculate(preNumber: String, postNumber: String): String {
+            if (!preNumber.isNum() || !postNumber.isNum()) {
+                throw IllegalArgumentException("Parameter must be Number")
+            }
+            return (preNumber.toInt() * postNumber.toInt()).toString()
+        }
+    },
+    DIVIDE("/") {
+        override fun calculate(preNumber: String, postNumber: String): String {
+            if (!preNumber.isNum() || !postNumber.isNum()) {
+                throw IllegalArgumentException("Parameter must be Number")
+            }
+            return (preNumber.toInt() / postNumber.toInt()).toString()
+        }
+    };
+
+    abstract fun calculate(preNumber: String, postNumber: String): String
+
+    companion object {
+
+        fun getOperator(operatorInput: String): Operator {
+            val operator = findOperator(operatorInput)
+            if (operator != null) {
+                return operator
+            } else {
+                throw IllegalArgumentException("Cannot find operator")
+            }
+        }
+        private fun findOperator(operator: String): Operator? = values().firstOrNull { it.operator == operator }
+    }
+}
+
+private fun String.isNum(): Boolean {
+    this.forEach {
+        char ->
+
+        val charConvertedToCode = char.code
+
+        if (charConvertedToCode > 57 || charConvertedToCode < 48) {
```

Comment:
```suggestion
        if (charConvertedToCode > '9' || charConvertedToCode < '0') {
```

꼭 아스키 코드로 적지 않아도 충분히 가능합니다..!

Reply:
추가적으로 계산기다보니 10보다 큰 숫자가 들어올 수 있는데요~
따라서 해당 함수의 로직을 다시 고민해보시면 좋을듯합니다!

Reply:
로직 수정해보았습니다.